src/Featurizers/TfidfVectorizerFeaturizer.h

```diff
@@ -0,0 +1,523 @@
+    using IndexMap                          = TfidfVectorizerTransformer::IndexMap;
+    using FrequencyMap                      = TfidfVectorizerTransformer::FrequencyMap;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    TfidfVectorizerEstimatorImpl(
+        AnnotationMapsPtr pAllColumnAnnotations,
+        size_t colIndex,
+        bool binary,
+        std::string norm,
+        bool use_idf,
+        bool smooth_idf,
+        bool sublinear_tf
+    );
+    ~TfidfVectorizerEstimatorImpl(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(TfidfVectorizerEstimatorImpl);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    size_t const                             _colIndex;
+    bool const                               _binary;
+    std::string const                        _norm;
+    bool const                               _use_idf;
+    bool const                               _smooth_idf;
+    bool const                               _sublinear_tf;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    bool begin_training_impl(void) override;
+
+    // MSVC has problems when the declaration and definition are separated
+    FitResult fit_impl(typename BaseType::InputType const *, size_t) override {
+        throw std::runtime_error("This should not be called");
+    }
+
+    void complete_training_impl(void) override;
+
+    // MSVC has problems when the declaration and definition are separated
+    typename BaseType::TransformerUniquePtr create_transformer_impl(void) override {
+
+        // ----------------------------------------------------------------------
+        using DocumentStatisticsAnnotationData          = Components::DocumentStatisticsAnnotationData;
+        using DocumentStatisticsEstimator               = Components::DocumentStatisticsEstimator<MaxNumTrainingItemsV>;
+
+        // ----------------------------------------------------------------------
+
+        DocumentStatisticsAnnotationData const &        data(DocumentStatisticsEstimator::get_annotation_data(BaseType::get_column_annotations(), _colIndex, Components::DocumentStatisticsEstimatorName));
+
+
+        typename DocumentStatisticsAnnotationData::FrequencyAndIndexMap const &
+                                                        termFrequencyAndIndex(data.TermFrequencyAndIndex);
+        std::uint32_t const                             totalNumDocus(data.TotalNumDocuments);
+
+        FrequencyMap termFrequency;
```

Comment:
>termFrequency [](start = 21, length = 13)

Update these variable names to indicate that there are more than one value (make it plural).